src/main/kotlin/at/bitfire/dav4jvm/property/caldav/CalendarHomeSet.kt

```diff
@@ -8,13 +8,18 @@ package at.bitfire.dav4jvm.property.caldav
 
 import at.bitfire.dav4jvm.Property
 import at.bitfire.dav4jvm.property.webdav.HrefListProperty
+import java.util.LinkedList
 import org.xmlpull.v1.XmlPullParser
 
-class CalendarHomeSet: HrefListProperty() {
+data class CalendarHomeSet(
+    override val hrefs: LinkedList<String> = LinkedList<String>()
```

Comment:
Applies to all List/Set properties:

If we want to go to the next level and have really immutable property classes, we would have to make the lists/sets also immutable.

That would mean that we had to construct them with a (non-mutable) `List`. (Or with a mutable list, but expose only a non-mutable list for reading.)

Then we would really need the `construct` parameter in `abstract class HrefListProperty.Factory` so that we can call the constructor with the list.